server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -530,7 +611,16 @@ public synchronized IndexShard createShard(
                 }
             }
 
-            Directory directory = directoryFactory.newDirectory(this.indexSettings, path);
+            Directory directory = null;
+            if (FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX_SETTING) &&
+            // TODO : Need to remove this check after support for hot indices is added in Composite Directory
+                this.indexSettings.isStoreLocalityPartial()) {
+                Directory localDirectory = directoryFactory.newDirectory(this.indexSettings, path);
+                directory = new CompositeDirectory(localDirectory, remoteDirectory, fileCache);
```

Comment:
is `CompositeDirectory` only be created for remote backed indexes?

Reply:
Yes, the plan is to create only for remote backed indices only - both hot and warm.
But in this PR, we are only enabling it for warm. Will include hot while adding support for hot/warm migration.